sdks/python/apache_beam/dataframe/frames_test.py

```diff
@@ -131,7 +137,12 @@ def _run_test(
             This option should NOT be set to False in tests added for new
             operations if at all possible. Instead make sure the new operation
             produces the correct proxy. This flag only exists as an escape hatch
-            until existing failures can be addressed (BEAM-12379)."""
+            until existing failures can be addressed (BEAM-12379).
+        check_dtypes (bool): Whether or not to check the dtypes of the values.
+            The default is True. This option should be set to False for
+            non-deferred operations, where the dtype of the values in the proxy
+            are not known ahead of time.
```

Comment:
I think this is actually a little bit dangerous as documented. We don't want to support operations where the dtype of the values can't be known at construction time. It's important that we have an accurate proxy so we know what operations are valid on the output.

That being said there are cases (like in this operations) where int64 columns get coerced to float64 because of NaNs. The _best_ solution here  would be to have something like a "`lenient_dtype_check`" option, which just checks that numeric columns are still numeric between actual and proxy.

I know that might be a bit more involved than what you want to get into for this PR though - if that's too much could you just update the docstring to say this is for operations with int64 to float64 coercion issues, and maybe a TODO to add `lenient_dtype_check`?

Reply:
Makes sense. Removed `check_dtypes ` and changed to `lenient_dtype_check` now. I've also added logic to do smarter inference of the dtypes, so hopefully this can help reduce our usage of `lenient_dtype_check`, at least for this operation.